src/main/java/uk/gov/hmcts/reform/roleassignment/domain/service/getroles/RetrieveRoleAssignmentOrchestrator.java

```diff
@@ -45,6 +49,24 @@ public RetrieveRoleAssignmentOrchestrator(PersistenceService persistenceService,
             UUID.fromString(actorId));
     }
 
+    public ResponseEntity<Object> retrieveRoleAssignmentsByActorIdAndCaseId(String actorId, String caseId,
+                                                                            String roleType) {
+        if (StringUtils.isEmpty(actorId) && StringUtils.isEmpty(caseId)) {
+            throw new BadRequestException(V1.Error.INVALID_ACTOR_AND_CASE_ID);
+        }
+
+        if (StringUtils.isNotEmpty(actorId)) {
+            ValidationUtil.validateInputParams(UUID_PATTERN, actorId);
+        }
+        if (StringUtils.isNotEmpty(caseId)) {
+            ValidationUtil.validateCaseId(caseId);
+        }
+
+        List<RoleAssignment> assignmentList =
+            persistenceService.getAssignmentsByActorAndCaseId(actorId,caseId, roleType);
+        return ResponseEntity.status(HttpStatus.OK).body(assignmentList);
```

Comment:
pls check if we need to send the resource object.

Reply:
As discussed, keeping it as is for now.